auction/src/lib.rs

```diff
@@ -374,15 +552,24 @@ pub mod pallet {
 			Ok(().into())
 		}
 
-		#[pallet::weight(T::WeightInfo::buy_it_now())]
+		#[pallet::weight((
+			{
+				let s = Deadlines::<T>::get().0.len();
```

Comment:
We could save us some typing if we implement the `len()` method for DeadlineList struct.